### PR TITLE
fix: updateUserAttributeSetting during add/update/push flows

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes
- adds attr during amplify add / amplify update flows
- adds this attr during amplify pull / push / build workflow
- for console and studio builds , this will work fine as contents are pushed with force
- for cli Cx , if Cx doesn't update the auth category , it needs a force push to update the userPool with this attribute.

#### Description of how you validated changes
- updated unit tests
- Manually tested
- https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/9793/workflows/f25bee72-b282-451e-a01e-eb673df2c60c

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by danielleadams** [comment]
@akshbhu this needs a rebase

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-user-pool-group-stack-builder.ts [hunk 1]
```diff
@@ -392,5 +391,5 @@ export class AmplifyUserPoolGroupStackOutputs extends cdk.Stack implements Ampli
   }
 
   public renderCloudFormationTemplate =
-  (_: cdk.ISynthesisSession): string => JSON.stringify(this._toCloudFormation(), undefined, 2);
+  (__: cdk.ISynthesisSession): string => JSON.stringify(this._toCloudFormation(), undefined, 2);
```
Comment (jhockett) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-user-pool-group-stack-builder.ts:394]: Same comment here - use JSONUtilities from cli-core

File: packages/amplify-appsync-simulator/package.json [hunk 2]
```diff
@@ -90,4 +90,4 @@
       "node"
     ]
   }
-}
+}
```
Comment (jhockett) [inline packages/amplify-appsync-simulator/package.json:93]: Do you have a VSCode setting that is removing newlines at the end of JSON files? Ideally all text files should end in a newline

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts [hunk 3]
```diff
@@ -519,12 +523,14 @@ export class AmplifyAuthTransform extends AmplifyCategoryTransform {
...
-            type: 'CommaDelimitedList',
-          },
-          `${key}`,
-        );
+        if (key !== 'userAutoVerifiedAttributeUpdateSettings') {
+          this._authTemplateObj.addCfnParameter(
+            {
+              type: 'CommaDelimitedList',
+            },
+            `${key}`,
```
Comment (edwardfoyle) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-stack-transform.ts:531]: Assuming `key` is a string:
```suggestion
            key,
```

**Review by edwardfoyle — APPROVED** [review]

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts [hunk 4]
```diff
@@ -513,13 +521,13 @@ export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCog
...
-        !_.isEmpty(props.authProviders) &&
-        !(Object.keys(props.authProviders).length === 1 && props.authProviders[0] === 'accounts.google.com' && props.audiences)
+        props.authProviders
+        && !_.isEmpty(props.authProviders)
+        && !(Object.keys(props.authProviders).length === 1 && props.authProviders[0] === 'accounts.google.com' && props.audiences)
       ) {
         this.identityPool.supportedLoginProviders = cdk.Lazy.anyValue({
           produce: () => {
-            let supprtedProvider: any = {};
+            const supprtedProvider: any = {};
```
Comment (jhockett) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts:530]: nit: typo and use of `any`
```suggestion
            const supportedProvider: $TSObject = {};
```

**Review by danielleadams — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**